frontend/src/Onboarding/RedirectScreen.tsx

```diff
@@ -57,21 +62,32 @@ export const RedirectScreen: React.FC<Props> = ({ redirectUrl }) => {
           path: "/",
           domain: window.location.hostname,
         }); // set persisting cookie for all paths
-
-        fetchUser(cookie).then(response => {
-          dispatch(setFullNameAction(response.user.username));
-          dispatch(setGraduationYearAction(response.user.graduationYear));
-          dispatch(setAcademicYearAction(response.user.academicYear));
-          const maj = findMajorFromName(response.user.major, majors);
-          if (maj) {
-            dispatch(setDeclaredMajorAction(maj));
-          }
-          dispatch(setUserIdAction(response.user.id));
-          dispatch(setEmailAction(response.user.email));
-          dispatch(setUserCoopCycleAction(response.user.coopCycle));
-          dispatch(setIsAdvisorAction(response.user.isAdvisor));
-          setIsLoading(false);
-        });
+        fetchUser(cookie)
+          .then(response => {
+            dispatch(setFullNameAction(response.user.username));
+            dispatch(setGraduationYearAction(response.user.graduationYear));
+            dispatch(setAcademicYearAction(response.user.academicYear));
+            const maj = findMajorFromName(response.user.major, majors);
+            if (maj) {
+              dispatch(setDeclaredMajorAction(maj));
+            }
+            dispatch(setUserIdAction(response.user.id));
+            dispatch(setEmailAction(response.user.email));
+            dispatch(setUserCoopCycleAction(response.user.coopCycle));
+            dispatch(setIsAdvisorAction(response.user.isAdvisor));
+            getScheduleCoursesFromSimplifiedCourseDataAPI(
+              response.user.coursesCompleted
+            ).then(courses => {
+              dispatch(setCompletedCourses(courses));
+            });
+            getScheduleCoursesFromSimplifiedCourseDataAPI(
+              response.user.coursesTransfer
+            ).then(courses => {
+              dispatch(setCompletedCourses(courses));
+            });
+            setIsLoading(false);
```

Comment:
I think we might run into a race condition here. I think we can make use of `Promise.all` for the two requests, and then do `setIsLoading(false)` on resolve of the `Promise.all`